Pocket/pocket-ios
# IN-507-Implemented Sort Menu for My List - Saved Items only

## Summary
Implementation of Sort Menu on My List (for Saved Items only) for iPhone & iPad both

## References 
[IN-507](https://getpocket.atlassian.net/browse/IN-507)

## Implementation Details
Sort Menu is implement by using UITableView. 'ListOptions' class is introduced to persist 'selection of sort option' locally in device so it persist the user's last choice of sorting option for My List.

## Test Steps
when I view the My List tab and it shows saved items in All and Favorite.
then I should be able to click on 'Sort/Filter' menu and apply sorting by choosing 'Newest option'. Saved items should sort by 'Newest' saved items.

when I view the My List tab and it shows saved items in All and Favorite.
then I should be able to click on 'Sort/Filter' menu and apply sorting by choosing 'Oldest option'. Saved items should sort by 'Oldest' saved items.

when I view the My List tab and it shows saved items in All and Favorite.
I should be able to click on 'Sort/Filter' menu and apply sorting by choosing 'Newest option'. Saved items should sort by 'Newest' saved items.
then If I close or terminate the app and come back to My List, it should show the saved items list as per last selected sorting option and when tap on 'Sort/Filter', by default the last option should be selected/highlighted.

## PR Checklist:

## Screenshots
| Initial State (sort by: Newest)        | Sort Menu (by default: Newest)           | Sort Menu (selected: Oldest)  | List sorted by Oldest |
|-------------|-------------|-----|------|
| ![Screenshot 1](https://user-images.githubusercontent.com/110417980/186437324-0812959b-d64b-409b-a772-5940c82009b4.png) | ![Screenshot 2](https://user-images.githubusercontent.com/110417980/186437342-88fe2f2e-43e1-4ef9-a4d8-140593a34854.png) | ![Screenshot 3](https://user-images.githubusercontent.com/110417980/186437378-da2ce45e-ee6d-48fa-8d71-b637c6e7a4d0.png) | ![Screenshot 4](https://user-images.githubusercontent.com/110417980/186437396-ac1cf4c9-3dfb-4671-a75d-6ec38b0656a0.png) |
